packages/flutter_web_plugins/lib/src/navigation/url_strategy.dart

```diff
@@ -223,17 +221,17 @@ abstract class PlatformLocation {
   /// The `pathname` part of the URL in the browser address bar.
   ///
   /// See: https://developer.mozilla.org/en-US/docs/Web/API/Location/pathname
-  String get pathname;
+  String? get pathname;
 
   /// The `query` part of the URL in the browser address bar.
   ///
   /// See: https://developer.mozilla.org/en-US/docs/Web/API/Location/search
-  String get search;
+  String? get search;
```

Comment:
Looks like `HashUrlStrategy` doesn't deal well with null `pathname` and `search`. Do we need to allow nulls here?

Reply:
In this case, I tried to preserve the logic from dart:html; [`Location.search`][search] and [`Location.pathname`][pathname] are both defined as `String?`.

At first I thought it was because their behavior is not defined when they're missing, but pathname should return an empty string when undefined.

Anyway, I can modify this and make it non-null, and then ensure that we return the correct defaults in the specific implementations.

[search]: https://developer.mozilla.org/en-US/docs/Web/API/Location/search
[pathname]: https://developer.mozilla.org/en-US/docs/Web/API/Location/pathname

Reply:
I made all these non-null in the abstract class, and provided sane defaults in the specific implementation. The rest of the class works better now, less `!` thrown around.